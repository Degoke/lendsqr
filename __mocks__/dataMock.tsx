import UsersIcon from '../public/images/users_card_icon.svg'

export const mockCard = {
  image: UsersIcon,
  text: 'Users',
  amount: 100,
}

export const mockImage = UsersIcon

export const MockComponent = () => (
  <div data-testid="mockComponent">Hello World</div>
)

export const mockList = {
  title: 'settings',
  items: [
    {
      text: 'preferences',
      url: '',
      icon: UsersIcon,
    },
    {
      text: 'fees and pricing',
      url: '',
      icon: UsersIcon,
    },
    {
      text: 'audit logs',
      url: '',
      icon: UsersIcon,
    },
  ],
}
