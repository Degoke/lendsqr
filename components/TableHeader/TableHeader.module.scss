@use '../../styles/variables' as *;

.tableHeader {
  text-transform: uppercase;
  text-align: center;
  font-weight: 600;
}
