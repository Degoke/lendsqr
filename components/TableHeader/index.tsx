import Image from 'next/image'
import FilterIcon from '../../public/images/filter_icon.svg'
import styles from './TableHeader.module.scss'
type Props = {
  text: string
}

export default function TableHeader({ text }: Props) {
  return (
    <th className={styles.tableHeader}>
      <span>{text}</span>
      <Image src={FilterIcon} alt="filter" width={12} height={12} />
    </th>
  )
}
