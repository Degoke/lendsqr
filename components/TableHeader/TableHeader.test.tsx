import { render, screen } from '@testing-library/react'
import TableHeader from './index'
import { mockCard, mockImage } from '../../__mocks__/dataMock'

describe('TableHeader tests', () => {
  it('renders TableHeader component unchanged', () => {
    const { container } = render(<TableHeader text={''} />)
    expect(container).toMatchSnapshot()
  })
})
