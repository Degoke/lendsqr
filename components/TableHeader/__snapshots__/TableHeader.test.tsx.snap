// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeader tests renders TableHeader component unchanged 1`] = `
<div>
  <th
    class="tableHeader"
  >
    <span />
    <img
      alt="filter"
      data-nimg="1"
      decoding="async"
      height="12"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=32&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=16&q=75 1x, /_next/image?url=%2Fimg.jpg&w=32&q=75 2x"
      style="color: transparent;"
      width="12"
    />
  </th>
</div>
`;
